datadog/resource_datadog_cloud_configuration_rule.go

```diff
@@ -0,0 +1,296 @@
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The name of the cloud configuration rule.",
+		},
+		messageField: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The message associated to the rule that will be shown in findings and signals.",
+		},
+		enabledField: {
+			Type:        schema.TypeBool,
+			Required:    true,
+			Description: "Whether the cloud configuration rule is enabled.",
+		},
+		policyField: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Policy written in Rego format.",
+		},
+		resourceTypeField: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Main resource type to be checked by the rule.",
+		},
+		relatedResourceTypesField: {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "Related resource types to be checked by the rule. Defaults to empty list.",
+			MinItems:    0,
+			MaxItems:    9,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		},
+		severityField: {
+			Type:             schema.TypeString,
+			ValidateDiagFunc: validators.ValidateEnumValue(datadogV2.NewSecurityMonitoringRuleSeverityFromValue),
+			Required:         true,
+			Description:      "Severity of the rule and associated signals.",
+		},
+		notificationsField: {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "Notification targets for signals. Defaults to empty list.",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		},
+		groupByField: {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "Fields to group by when generating signals, e.g. @resource. Defaults to empty list.",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		},
+		tagsField: {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "Tags of the rule, propagated to findings and signals. Defaults to empty list.",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		},
+	}
+}
+
+func cloudConfigurationRuleCreateContext(ctx context.Context, d *schema.ResourceData, metadata interface{}) diag.Diagnostics {
+	providerConf := metadata.(*ProviderConfiguration)
+	apiInstances := providerConf.DatadogApiInstances
+	auth := providerConf.Auth
+
+	ruleCreate := buildRuleCreatePayload(d)
+
+	response, httpResponse, err := apiInstances.GetSecurityMonitoringApiV2().CreateSecurityMonitoringRule(auth, ruleCreate)
+	if err != nil {
+		return utils.TranslateClientErrorDiag(err, httpResponse, "error creating security monitoring rule")
+	}
+	if err := utils.CheckForUnparsed(response); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if response.SecurityMonitoringStandardRuleResponse != nil {
+		d.SetId(response.SecurityMonitoringStandardRuleResponse.GetId())
+		updateResourceDataFromResponse(d, response.SecurityMonitoringStandardRuleResponse)
+	} else {
+		return diag.FromErr(fmt.Errorf("SecurityMonitoringStandardRuleResponse is empty"))
+	}
+
+	return nil
```

Comment:
I think making the error check is clearer wdyt?
```suggestion
	if response.SecurityMonitoringStandardRuleResponse == nil {
		return diag.FromErr(fmt.Errorf("SecurityMonitoringStandardRuleResponse is empty"))
	}

	d.SetId(response.SecurityMonitoringStandardRuleResponse.GetId())
	updateResourceDataFromResponse(d, response.SecurityMonitoringStandardRuleResponse)
	return nil
```